### PR TITLE
Sink: pass ownership of the packet on write()

### PR DESCRIPTION
**Pull request by gdesmott**
Prevent a copy if the implementation needs to keep the data around.

### CODE REVIEW

**Comment by roderickvd** [comment]
What's the rationale behind this? Is it about ergonomics (ownership for someone else's code) or performance (preventing copying)?

If I understand correctly, we were first passing packets by reference, then copying samples. Your suggestion seems to be to copy packets, then pass samples by reference. Performance-wise, I think it won't matter much?

**Comment by gdesmott** [comment]
> What's the rationale behind this? Is it about ergonomics (ownership for someone else's code) or performance (preventing copying)?

Performance, my gstreamer element needs to copy each packet which is suboptimal.

**Comment by roderickvd** [comment]
I'm OK with this since it won't hurt the performance of the `librespot` player but can benefit others. It will be a small breaking change for others. Can you update `CHANGELOG.md`?

**Comment by gdesmott** [comment]
> Can you update `CHANGELOG.md`?

done.

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - [playback] `alsa`: Improve `--device ?` functionality for the alsa backend.
 - [contrib] Hardened security of the systemd service units
 - [main] Verbose logging mode (`-v`, `--verbose`) now logs all parsed environment variables and command line arguments (credentials are redacted).
+- [playback] `Sink`: `write()` now receives owernship of the packet.
```
Comment (roderickvd) [inline CHANGELOG.md:16]: Please suffix this with `(breaking)`. Also a small typo.

**Review by roderickvd — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (gdesmott) [reply CHANGELOG.md]: done.

**Review by roderickvd — APPROVED** [review]

**Pull request merged**